sdk/keyvault/Azure.Security.KeyVault.Administration/readme.md

```diff
@@ -1,34 +1,194 @@
+    ```PowerShell
+    $Env:AZURE_CLIENT_ID="generated-app-ID"
+    $Env:AZURE_CLIENT_SECRET="random-password"
+    $Env:AZURE_TENANT_ID="tenant-ID"
+    ```
+
+* Grant the above mentioned application authorization to perform key operations on the Azure Key Vault:
+    ```PowerShell
+    az keyvault set-policy --name <your-key-vault-name> --spn $AZURE_CLIENT_ID --key-permissions backup delete get list create encrypt decrypt update
+    ```
+    > --key-permissions:
+    > Accepted values: backup, create, decrypt, delete, encrypt, get, import, list, purge, recover, restore, sign, unwrapKey, update, verify, wrapKey
+
+* Use the above mentioned Azure Key Vault name to retrieve details of your Vault which also contains your Azure Key Vault URL:
+    ```PowerShell
+    az keyvault show --name <your-key-vault-name>
+    ```
+
+#### Create KeyVaultAccessControlClient
+Once you've populated the **AZURE_CLIENT_ID**, **AZURE_CLIENT_SECRET** and **AZURE_TENANT_ID** environment variables and replaced **your-vault-url** with the above returned URI, you can create the [KeyVaultAccessControlClient][rbac_client]:
+
+```C# Snippet:CreateKeyClient
+// Create a new access control client using the default credential from Azure.Identity using environment variables previously set,
+// including AZURE_CLIENT_ID, AZURE_CLIENT_SECRET, and AZURE_TENANT_ID.
+var client = new KeyVaultAccessControlClient(vaultUri: new Uri(keyVaultUrl), credential: new DefaultAzureCredential());
+
+// Retrieve all the role definitions.
+List<RoleDefinition> roleDefinitions = client.GetRoleDefinitions(RoleAssignmentScope.Global).ToList();
+
+// Retrieve all the role assignments.
+List<RoleDefinition> roleAssignments = client.GetRoleAssignments(RoleAssignmentScope.Global).ToList();
+
+```
 
 ## Key concepts
 
-Content forthcoming
+### RoleDefinition
+A `RoleDefinition` is a collection of permissions. A role definition defines the operations that can be performed, such as read, write, and delete. It can also define the operations that are excluded from allowed operations.
+
+RoleDefinitions can be listed and specified as part of a `RoleAssignment`.
+
+### RoleAssignment. 
+A `RoleAssignment` is the association of a RoleDefinition to a service principal. They can be created, listed, fetched individually, and deleted.
+
+### KeyVaultAccessControlClient
+A `KeyVaultAccessControlClient` provides both synchronous and asynchronous operations allowing for listing management of `RoleDefinition` and `RoleAssignment` objects.
 
 ## Examples
+The Azure.Security.KeyVault.Administration package supports synchronous and asynchronous APIs.
 
-Content forthcoming
+The following section provides several code snippets using the `client` [created above](#create-keyvaultaccesscontrolclient), covering some of the most common Azure Key Vault access control related tasks:
+
+### List the role definitions
+List the role definitions available for assignment.
+
+```C# Snippet:GetRoleDefinitions
```

Comment:
Where is the code for these snippets? Was it already in place (I don't recall seeing it). Do not write code inline. It should be in a sample that we actually run live. We may need to prevent them from running live currently given availability of the new features, but we still want samples that compile to make sure that, as we refactor or whatever, the code is correct.

When Key Vault was still a track 2 preview SDK, we had many samples that didn't even compile nonetheless run. Several of us worked to define this new structure we use now to prevent that from happening again.

Reply:
Ah, I didn’t realize that all these example snippets should be auto-generated from full samples. I’ll get the samples added soon.